spec/OrderProcessor/OrderTaxesProcessorSpec.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\RefundPlugin\OrderProcessor;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Addressing\Matcher\ZoneMatcherInterface;
+use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Core\Model\AddressInterface;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\Scope;
+use Sylius\Component\Core\Provider\ZoneProviderInterface;
+use Sylius\Component\Core\Taxation\Exception\UnsupportedTaxCalculationStrategyException;
+use Sylius\Component\Core\Taxation\Strategy\TaxCalculationStrategyInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+use Sylius\Component\Registry\PrioritizedServiceRegistryInterface;
+use Sylius\RefundPlugin\Entity\ShipmentInterface;
+
+final class OrderTaxesProcessorSpec extends ObjectBehavior
+{
+    function let(
+        ZoneProviderInterface $defaultTaxZoneProvider,
+        ZoneMatcherInterface $zoneMatcher,
+        PrioritizedServiceRegistryInterface $strategyRegistry
+    ): void {
+        $this->beConstructedWith($defaultTaxZoneProvider, $zoneMatcher, $strategyRegistry);
+    }
+
+    function it_is_an_order_processor(): void
+    {
+        $this->shouldImplement(OrderProcessorInterface::class);
+    }
+
+    function it_processes_taxes_using_a_supported_tax_calculation_strategy(
+        ZoneMatcherInterface $zoneMatcher,
+        PrioritizedServiceRegistryInterface $strategyRegistry,
+        OrderInterface $order,
+        OrderItemInterface $orderItem,
+        ShipmentInterface $shipment,
+        AddressInterface $address,
+        ZoneInterface $zone,
+        TaxCalculationStrategyInterface $strategyOne,
+        TaxCalculationStrategyInterface $strategyTwo
+    ): void {
+        $order->getItems()->willReturn(new ArrayCollection([$orderItem->getWrappedObject()]));
+        $order->getShipments()->willReturn(new ArrayCollection([$shipment->getWrappedObject()]));
+        $order->isEmpty()->willReturn(false);
+        $order->getBillingAddress()->willReturn($address);
+
+        $order->removeAdjustments(AdjustmentInterface::TAX_ADJUSTMENT)->shouldBeCalled();
+        $orderItem->removeAdjustmentsRecursively(AdjustmentInterface::TAX_ADJUSTMENT)->shouldBeCalled();
```

Comment:
```suggestion
        $order->getItems()->willReturn(new ArrayCollection([$orderItem->getWrappedObject()]));
        $order->getShipments()->willReturn(new ArrayCollection([$shipment->getWrappedObject()]));
        $order->isEmpty()->willReturn(false);
        $order->getBillingAddress()->willReturn($address);
        $order->removeAdjustments(AdjustmentInterface::TAX_ADJUSTMENT)->shouldBeCalled();
        
        $orderItem->removeAdjustmentsRecursively(AdjustmentInterface::TAX_ADJUSTMENT)->shouldBeCalled();
```

Here and below

Reply:
It is `shouldBeCalled`, not `willReturn`, I wanted to group it with other `shouldBeCalled` statements.